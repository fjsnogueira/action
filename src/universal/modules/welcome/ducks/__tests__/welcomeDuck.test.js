import reducer, {
  goToPage,
  setWelcomeTeam,
  nextPage,
  previousPage,
  updateCompleted
} from '../welcomeDuck';

test('initial state', () => {
  expect(reducer()).toMatchSnapshot();
});

test('setWelcomeTeam() updates teamId, teamMemberId', () => {
  expect(
    reducer(undefined,
      setWelcomeTeam({
        teamId: 'apple1',
        teamMemberId: 'banana2'
      })
    )
  ).toMatchSnapshot();
});

test('nextPage() increments', () => {
  const initialState = reducer();
  expect(reducer(initialState, nextPage()))
    .toMatchSnapshot();
});

test('nextPage() does not exceed 3', () => {
  let state = reducer();
  for (let i = 0; i < 4; i++) {
    state = reducer(state, nextPage());
  }

  expect(reducer(state, nextPage())).toMatchSnapshot();
});

test('previousPage() decrements', () => {
  let state = reducer();
  state = reducer(state, nextPage());

  expect(reducer(state, previousPage())).toMatchSnapshot();
});

test('previousPage() is never less than one', () => {
  let state = reducer();
  state = reducer(state, previousPage());
  expect(reducer(state, previousPage())).toMatchSnapshot();
});

test('updateCompleted() updates state', () => {
  const state = reducer();
  expect(reducer(state, updateCompleted(2))).toMatchSnapshot();
});

test('goToPage', () => {
  expect(reducer(undefined, goToPage(1))).toMatchSnapshot();
});
