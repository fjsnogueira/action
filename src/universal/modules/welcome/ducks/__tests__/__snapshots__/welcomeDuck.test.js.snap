// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goToPage 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 1,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`initial state 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 1,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`nextPage() does not exceed 3 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 3,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`nextPage() increments 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 2,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`previousPage() decrements 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 1,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`previousPage() is never less than one 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 1,
  "teamId": null,
  "teamMemberId": null,
}
`;

exports[`setWelcomeTeam() updates teamId, teamMemberId 1`] = `
Object {
  "completed": 0,
  "existingInvites": Array [],
  "page": 1,
  "teamId": "apple1",
  "teamMemberId": "banana2",
}
`;

exports[`updateCompleted() updates state 1`] = `
Object {
  "completed": 2,
  "existingInvites": Array [],
  "page": 1,
  "teamId": null,
  "teamMemberId": null,
}
`;
